dlib/statistics/statistics.h

```diff
@@ -22,43 +22,24 @@ namespace dlib
     {
     public:
 
-        running_stats()
-        {
-            clear();
-
-            COMPILE_TIME_ASSERT ((
-                    is_same_type<float,T>::value ||
-                    is_same_type<double,T>::value ||
-                    is_same_type<long double,T>::value 
-            ));
-        }
+        static_assert(std::is_floating_point<T>::value, "T must be a floating point type");
 
         void clear()
         {
-            sum_     = 0;
-            sum_sqr  = 0;
-            sum_cub  = 0;
-            sum_four = 0;
-
-            n = 0;
-            min_value = std::numeric_limits<T>::infinity();
-            max_value = -std::numeric_limits<T>::infinity();
+            running_stats empty{};
+            std::memcpy(this, &empty, sizeof(empty));
         }
 
         void add (
-            const T& val
+            T val
```

Comment:
Did removing the const reference change the result?

Reply:
So the reason I did that is because all pointers (and references) take 8 bytes. A float value is 4 and double is 8. So you're better off using values. Also value semantics are better in general.

Reply:
Yeah that's fine.  Was just wondering if that was somehow part of the trigger to make it wonky.